tests/stress/deployments.py

```diff
@@ -0,0 +1,761 @@
+        self._set_k8s_context()
+        print("Set K8s context!")
+
+        # Prepare the Helm command
+        helm_cmd = ["helm", "install", self.release_name, self.chart_name, "-f", self.values_path]
+
+        # Execute the Helm command to install Memgraph
+        subprocess.run(helm_cmd, check=True)
+        print("Helm charts installed!")
+
+        # Wait for Memgraph to be fully deployed
+        self.wait_for_server()
+        print("Server ready!")
+
+    def stop_memgraph(self) -> None:
+        """Uninstalls Memgraph using Helm"""
+        helm_cmd = ["helm", "uninstall", self.release_name]
+        subprocess.run(helm_cmd, check=True)
+
+    def cleanup(self):
+        # Delete Minikube cluster after cleanup
+        self.delete_minikube()
+
+    def wait_for_server(self) -> None:
+        """Wait for Memgraph pod to be running and ready"""
+        print("Waiting for Memgraph pod to be running and ready...")
+        while True:
+            try:
+                # Get the pod name
+                pod_name_cmd = [
+                    "kubectl",
+                    "get",
+                    "pods",
+                    "-l",
+                    f"app.kubernetes.io/instance={self.release_name}",
+                    "-o",
+                    "jsonpath='{.items[0].metadata.name}'",
+                ]
+                pod_name = subprocess.check_output(pod_name_cmd).decode("utf-8").strip().strip("'")
+
+                # Get the pod status
+                pod_status_cmd = ["kubectl", "get", "pod", pod_name, "-o", "jsonpath='{.status.phase}'"]
+                pod_status = subprocess.check_output(pod_status_cmd).decode("utf-8").strip().strip("'")
+
+                # Get the container readiness (1/1)
+                container_status_cmd = [
+                    "kubectl",
+                    "get",
+                    "pod",
+                    pod_name,
+                    "-o",
+                    "jsonpath='{.status.containerStatuses[0].ready}'",
+                ]
+                container_status = subprocess.check_output(container_status_cmd).decode("utf-8").strip().strip("'")
+
+                if pod_status == "Running" and container_status == "true":
+                    print(f"Pod {pod_name} is running and ready!")
+                    break
+                else:
+                    print(f"Pod {pod_name} not ready yet, status: {pod_status}, container ready: {container_status}")
+                    time.sleep(5)
+            except Exception as e:
+                print(f"Pod not ready yet!")
+                time.sleep(1)
+
+    def execute(self, query: str) -> None:
+        """Execute a Cypher query on Memgraph"""
+        # Get Minikube IP and NodePort
+        minikube_ip = self.get_minikube_ip()
+        if not minikube_ip:
+            print("Failed to get Minikube IP.")
+            return
+
+        nodeport = self.get_service_nodeport(self.release_name)
```

Comment:
This is good but NodePort service probably won't be used that much... BDO uses IngressNginx. You will want to have different methods for different services.